.github/copilot-instructions.md

```diff
@@ -0,0 +1,204 @@
+
+### I2C Sensors
+- `dht(name, pin, type)` - DHT11/DHT22 temperature/humidity
+- `i2c_bmp180(name)` - Barometer/temperature
+- `i2c_bh1750(name)` - Light sensor
+- `ds18b20(name, pin)` - OneWire temperature
+
+### RGB/LED Strips
+- `rgb_single(name, pin_r, pin_g, pin_b)` - Single RGB LED
+- `rgb_strip(name, pin, led_count)` - WS2812/NeoPixel strips
+
+See `/doc/node_help/` for complete device documentation.
+
+## Build and Deploy Process
+
+### Compile Only
+```bash
+cd ~/iot-systems/my-home/node1
+compile  # or: deploy compile
+```
+
+### Deploy via Serial
+```bash
+deploy serial  # Auto-detects USB port
+# or specify: deploy serial /dev/ttyUSB0
+```
+
+### Deploy Over-The-Air (OTA)
+```bash
+deploy  # Automatically uses OTA if node is online
+```
+
+### Adopting Existing Nodes
+For nodes already flashed with IoTempower:
+```bash
+cd node-folder
+adopt  # or: initialize
+# Follow prompts to configure WiFi and complete setup
+```
+
+## Important Coding Guidelines
+
+### When Modifying Device Drivers (`/lib/node_types/esp/src/`)
+1. All devices inherit from `Device` base class
+2. Override `start()` for initialization, `measure()` for polling
+3. Use `changed()` to mark when values should be published
+4. Add device metadata to `devices.ini` for dependency tracking
+5. Test with multiple board types (esp8266, esp32)
+
+### When Modifying Build Scripts (`/bin/`)
+1. Maintain compatibility with both Raspberry Pi image and standalone Linux
+2. Scripts must work from both node folders and system folders
+3. Use `IOTEMPOWER_ROOT` and `IOTEMPOWER_ACTIVE` environment variables
+4. Handle errors gracefully (users may not have all dependencies)
+
+### When Adding Examples
+1. Create folder with descriptive name in `/examples/`
+2. Include `node.conf` specifying board type
+3. Add well-commented `setup.cpp` showing device usage
+4. Document pin connections in comments or README
+
+## Testing Changes
+
+### Before Committing
+1. Compile test nodes: `cd examples/plush-owl && compile`
+2. If possible, test deployment on actual hardware
+3. Verify documentation builds: `cd doc && make html`
+4. Check that changes work on both ESP8266 and ESP32 boards
+
+## Reference Documentation
+
+**Must read before significant changes:**
+- `/doc/architecture.rst` - Complete system architecture
+- `/doc/first-node.rst` - Basic workflow and concepts
+- `/doc/second-node.rst` - OTA deployment patterns
+- `/doc/device-architecture.rst` - Device framework internals
+- `/doc/deployment-process.rst` - Build system details
+
+## Quick Tips
+
+- Device driver code is in `/lib/node_types/esp/src/dev_*.cpp`
+- User-facing documentation is in `/doc/*.rst` (ReStructuredText)
+- System commands are in `/bin/` (mostly Bash scripts)
+- Example configurations are in `/examples/`
+- PlatformIO configuration is in `/lib/node_types/esp/platformio.ini`
+- The `build/` folder in each node is transient - never edit it directly
+- MQTT topics follow pattern: `<system>/<node>/<device>`
```

Comment:
no, not <system> but folder hierarchy under system - change origin of this information (if that has been documented by mistake, please change that in the respective documentation)